lib/unison-sqlite/src/Unison/Sqlite/Connection.hs

```diff
@@ -0,0 +1,481 @@
+module Unison.Sqlite.Connection
+  ( -- * Connection management
+    Connection (..),
+    withConnection,
+    openConnection,
```

Comment:
Let's not even export `openConnection` and `closeConnection`; everyone should always use `withConnection`. We've already seen they're too error-prone.

Reply:
I agree. Note that the main interface, `Unison.Sqlite`, only exports `withConnection`. But I figured certain patterns, like using the `Acquire` applicative, require access to the acquire/release actions separately.

Reply:
I'd much prefer we leave them unexported until that need comes up. It's easy to add an export and mark it as `unsafe` in the future if we discover we really need it.

What use-cases are there for Acquire that aren't covered by `withTransaction`? They both seem to have similar interfaces and both work within MonadUnliftIO unless I'm mistaken.

Reply:
Ok, I hid the exports. `Acquire` is just a different style of resource acquisition. You might use it to acquire many things in sequence and yet only incur one syntactic level of indentation.